code/modules/cm_marines/equipment/sentries.dm

```diff
@@ -123,6 +123,9 @@
 	if(do_after(user, 40, TRUE, 5, BUSY_ICON_BUILD))
 		if(!src || anchored)//Check if we got exploded
 			return
+		if(!Adjacent(user)) // We've moved away from the one picking us up.
```

Comment:
This is just patching a symptom of a greater problem. do_after should inherently have an adjacency check.